chain/messagepool/repub.go

```diff
@@ -20,17 +20,18 @@ const repubMsgLimit = 30
 var RepublishBatchDelay = 100 * time.Millisecond
 
 func (mp *MessagePool) republishPendingMessages(ctx context.Context) error {
-	mp.curTsLk.Lock()
+	mp.curTsLk.RLock()
 	ts := mp.curTs
 
 	baseFee, err := mp.api.ChainComputeBaseFee(context.TODO(), ts)
+	mp.curTsLk.RUnlock()
```

Comment:
Move this up one more line? We don't need to hold the curTsLk while base fee is computed over the API.

Reply:
Added this note to #3 on the follow on PR - https://github.com/filecoin-project/lotus/issues/10597